src/sentry/utils/appleconnect/appstore_connect.py

```diff
@@ -79,127 +79,100 @@ def _get_next_page(response_json: Mapping[str, Any]) -> Optional[str]:
-                "version": safe.get_path(build, "attributes", "version"),
-                "id": safe.get_path(build, "id"),
-            }
-            versions.append(b)
-        result.append(v)
-
-    return result
 
-
-def get_release_version_info(
-    session: Session, credentials: AppConnectCredentials, app_id: str
-) -> List[Dict[str, Any]]:
-    """Get all release builds version information for an application
-
-    The release build version information has the following structure:
-    platform: str - the platform for the build (e.g. IOS, MAC_OS ...)
-    short_version: str - the short version build info ( e.g. '1.0.1'), also called "train"
-       in starship documentation
-    id: str - the IID of the version
-    versions: vec - a vector with builds
-       version: str - the version of the build (e.g. '101'), looks like the build number
-       id: str - the IID of the build
-
-    NOTE: the release version information is identical to the pre release version information
-    :return: a list of release builds version information (see above)
-    """
-    url = f"v1/apps/{app_id}/appStoreVersions"
-    data = _get_appstore_info_paged_data(session, credentials, url)
     result = []
-    for d in data:
-        versions: List[Dict[str, Any]] = []
-        build_url = safe.get_path(d, "relationships", "build", "links", "related")
-        v = {
-            "platform": safe.get_path(d, "attributes", "platform"),
-            "short_version": safe.get_path(d, "attributes", "versionString"),
-            "id": safe.get_path(d, "id"),
-            "versions": versions,
-        }
-
-        build_info = _get_appstore_info_paged_data(session, credentials, build_url)
-        build_info = safe.get_path(build_info, "data")
-        if build_info is not None:
-            # Note RaduW: never seen info in this structure, I assume the same structure as pre release
-            versions.append(
-                {
-                    "id": safe.get_path(build_info, "id"),
-                    "version": safe.get_path(build_info, "attributes", "version"),
-                }
-            )
-        result.append(v)
-    return result
 
+    url = (
+        f"v1/builds?filter[app]={app_id}"
+        # we can fetch a maximum of 200 builds at once, so do that
+        "&limit=200"
+        # include related AppStore/PreRelease versions with the response
+        # TODO: the `iris` web API has related `buildBundles` objects,
+        # which have very useful `includesSymbols` and `dSYMUrl` attributes,
+        # but this is sadly not available in the official API. :-(
+        "&include=appStoreVersion,preReleaseVersion"
+        # sort newer releases first
+        "&sort=-uploadedDate"
+        # TODO: figure out if we actually can/want to download invalid or expired builds
+        # only include valid builds
+        "&filter[processingState]=VALID"
+        # and builds that have not expired yet
+        "&filter[expired]=false"
```

Comment:
I’m a bit unsure about this here. It does reduce the number of builds that we have to enumerate here by a lot, but might miss builds that have expired but might still be in use?